### DISCUSSION TITLE
Add support for e.g. `psyclone help OMPParallelLoopTrans`

### DISCUSSION THREAD

**Discussion by arporter** [stfc/PSyclone#2713] [Ideas]
I think it would be useful if, like e.g. git, we could support a user querying how to use a particular transformation. For this to be useful, we'd need to somehow combine the docstrings of the class itself and those classes from which it inherits. e.g. OMPParallelLoopTrans inherits from ParallelLoopTrans and it is that class that has the "ignore_dependencies_for" option. I don't know how we'd do that.

**Comment by arporter** [comment]
You can do something like:
![image](https://github.com/user-attachments/assets/df02bf2b-9f5a-46c2-bbcb-af59a3b82335)
Question is, can we get Sphinx to render our docstring in a pretty way in the terminal?

**Reply by sergisiso** [reply]
You can do that for any python file with pydoc :)

`python -m pydoc src/psyclone/psyir/transformations/parallel_loop_trans.py`

**Reply by sergisiso** [reply]
docstring including the super docstring (or part of it) has been recently discussed here: https://github.com/stfc/PSyclone/pull/2692#discussion_r1744024669

If that's something we want I can promote it to an issue

**Reply by arporter** [reply]
Ooh, pydoc is nice, I didn't know about that. I was just struck (while working on my slides) by how hard it can be to work out what options there are to a Transformation. I think it would be valuable if we could make this easier.

**Reply by sergisiso** [reply]
To be clear, I do not disagree with a `psyclone help` interface, just that the info should be what is already in the docstrings so people using other tools: pydoc, lsp, ... also see the full information.

Another aspect that I find that is not very discoverable is the available transformations. At some point we could recover the transformation listing functionality that is now broken. And would be convenient to have a command flag or part of this help command that could list them.

**Reply by sergisiso** [reply]
There is also #2668 which questions if Transformation Option would be better documented if they were kwargs

**Reply by hiker** [reply]
I agree some easy help mode would be great.

> Another aspect that I find that is not very discoverable is the available transformations. At some point we could recover the transformation listing functionality that is now broken. And would be convenient to have a command flag or part of this help command that could list them.

Yes, that is really difficult, esp. atm where transformations are in different paths (hopefully that will solve itself over time). Somewhat related (see https://github.com/stfc/PSyclone/issues/1153#issuecomment-795089856): how to know if there is a domain-specific transformation to use, or just the generic one.
